metadata/utils.go

```diff
@@ -250,16 +251,18 @@ func resolveAliases(md map[string]string, result any) error {
 }
 
 func toTruthyBoolHookFunc() mapstructure.DecodeHookFunc {
+	boolPtrType := reflect.TypeOf(ptr.Of(true))
+
 	return func(
 		f reflect.Type,
 		t reflect.Type,
 		data any,
 	) (any, error) {
-		if f == reflect.TypeOf("") && t == reflect.TypeOf(true) {
+		if f.Kind() == reflect.String && t.Kind() == reflect.Bool {
```

Comment:
I remember there being a reason why we couldn't use kind. Anyways, make sure this doesn't introduce a bug. Don't refactor that which isn't broken.

Reply:
I'll revert this